packages/react-components/react-nav-preview/src/components/NavContext.ts

```diff
@@ -0,0 +1,30 @@
+import { createContext, useContextSelector } from '@fluentui/react-context-selector';
+import type { Context, ContextSelector } from '@fluentui/react-context-selector';
+import { NavContextValue } from './NavContext.types';
+
+const navContextDefaultValue: NavContextValue = {
+  reserveSelectedNavGroupSpace: true,
+  selectedValue: undefined,
+  onRegister: () => {
+    /* noop */
+  },
+  onUnregister: () => {
+    /* noop */
+  },
+  onSelect: () => {
+    /* noop */
+  },
+  getRegisteredNavGroups: () => {
+    return {
+      registeredNavGroups: {},
+    };
+  },
+};
+
+export const NavContext: Context<NavContextValue> = createContext<NavContextValue | undefined>(
+  undefined,
+) as Context<NavContextValue>;
+
+export const NavProvider = NavContext.Provider;
+export const useNavContext = <T>(selector: ContextSelector<NavContextValue, T>): T =>
```

Comment:
this one should also have the unstable suffix

```suggestion
export const useNavContext_unstable = <T>(selector: ContextSelector<NavContextValue, T>): T =>
```